ercc/registry/registry.go

```diff
@@ -0,0 +1,255 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+Copyright 2020 Intel Corporation
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package registry
+
+import (
+	"encoding/base64"
+	"fmt"
+	"log"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/hyperledger-labs/fabric-private-chaincode/ercc/attestation"
+	"github.com/hyperledger-labs/fabric-private-chaincode/internal/protos"
+	"github.com/hyperledger-labs/fabric-private-chaincode/internal/utils"
+	"github.com/hyperledger/fabric-contract-api-go/contractapi"
+	"github.com/pkg/errors"
+)
+
+type Contract struct {
+	contractapi.Contract
+
+	Verifier   attestation.VerifierInterface
+	PEvaluator utils.PolicyEvaluatorInterface
+}
+
+// returns a set of credentials registered for a given chaincode id
+// Note: to get the endpoints of FPC endorsing peers do the following:
+// - discover all endorsing peers (and their endpoints) for the FPC chaincode using "normal" lifecycle
+// - query `getEnclaveId` at all the peers discovered
+// - query `queryListEnclaveCredentials` with all received enclave_ids
+// this gives you the endpoints and credentials including enclave_vk, and chaincode_ek
+func (rs *Contract) QueryListEnclaveCredentials(ctx contractapi.TransactionContextInterface, chaincodeId string) ([][]byte, error) {
```

Comment:
so by default the contract API only support a bunch of input and output primitive data types. Our credential data type is not supported. There is a way to provide custom serializer to make this possible but I haven't yet looked into it. 

Also, maybe we also don't care in this function because getState(ByPar...) returns a serialized Credentials object. So if this function would return an array of []Credentials. we would need to deserialize them after getState and returns. However, the custom serializer would serialize them again. Let me add more comments to the code.

Reply:
ok, completely missed when looking at it that these parameters come from the request, not one of our other go functions, and hence are the encoded version of these objects (i.e., serialized protobuf and as discussed in this morning calls also base64 encoded (on top) so on the client side we have strings which can be handled by peer cli script.
Clarifying that in documentation though would help. => all add a corresponding proposal for a change ...